src/test/java/com/example/moviemanager/service/MovieServiceTest.java

```diff
@@ -26,122 +26,122 @@
-        doThrow(EmptyResultDataAccessException.class).when(movieRepository).deleteById(movieId);
+        when(movieRepository.findById(ID_1)).thenReturn(Optional.empty());
 
         // then
         assertThatExceptionOfType(MovieNotFoundException.class)
-                .isThrownBy(() -> movieService.deleteById(movieId))
-                .withMessage("Deletion failed. Could not find movie with id - " + movieId);
+                .isThrownBy(() -> movieService.findById(ID_1))
+                .withMessage("Could not find movie by id - " + ID_1);
     }
 
     @Test
     void deletes_movie_by_id() {
         // given
-        long movieId = 1;
-        doNothing().when(movieRepository).deleteById(movieId);
+        doNothing().when(movieRepository).deleteById(ID_1);
 
         // when
-        movieService.deleteById(movieId);
+        movieService.deleteById(ID_1);
 
         // then
-        verify(movieRepository, times(1)).deleteById(movieId);
+        verify(movieRepository, times(1)).deleteById(ID_1);
         verifyNoMoreInteractions(movieRepository);
     }
 
+    @Test
+    void throws_exception_when_trying_to_delete_non_existing_movie() {
+        // given
+        doThrow(EmptyResultDataAccessException.class).when(movieRepository).deleteById(ID_1);
+
+        // then
+        assertThatExceptionOfType(MovieNotFoundException.class)
+                .isThrownBy(() -> movieService.deleteById(ID_1))
+                .withMessage("Deletion failed. Could not find movie with id - " + ID_1);
+    }
+
     @Test
     void saves_movie() {
         // given
-        Movie expectedMovie =
-                new Movie(1L, "Home Alone", "Christmas movie", 1990, 8.5);
-        when(movieRepository.save(expectedMovie)).thenReturn(expectedMovie);
+        Movie givenMovie = new Movie("Home Alone", "Christmas movie", 1990, 8.5);
+        Movie expectedMovie = new Movie(ID_1, "Home Alone", "Christmas movie", 1990, 8.5);
+        when(movieRepository.save(givenMovie)).thenReturn(expectedMovie);
 
         // when
-        Movie actualMovie = movieService.save(expectedMovie);
+        Movie actualMovie = movieService.save(givenMovie);
 
         // then
-        assertThat(expectedMovie.equals(actualMovie));
+        assertThat(actualMovie)
+                .isEqualTo(expectedMovie);
     }
 
     @Test
     void updates_movie_by_id_with_provided_data() {
         // given
-        long movieId = 1;
-        when(movieRepository.findById(movieId)).thenReturn(
-                Optional.of(new Movie(movieId, "Home Alone 2", "Comedy", 1992, 9.0)));
-        Movie expectedMovie =
-                new Movie(5L, "Home Alone", "Christmas movie", 1990, 8.5);
-        expectedMovie.setId(movieId);
+        when(movieRepository.findById(ID_1))
+                .thenReturn(Optional.of(new Movie(ID_1, "Home Alone 2", "Comedy", 1992, 9.0)));
+        Movie expectedMovie = new Movie("Home Alone", "Christmas movie", 1990, 8.5);
+        expectedMovie.setId(ID_1);
```

Comment:
Wanted to simulate same behavior as in update() method in MovieService class. Because service from controller receives movie without id, and should set it before passing it to repository.

But maybe I don`t need to expose all details in test.